ext/bcmath/tests/bcfloor.phpt

```diff
@@ -0,0 +1,46 @@
+--TEST--
+bcfloor() function
+--EXTENSIONS--
+bcmath
+--FILE--
+<?php
+$nums = [
+    '0',
+    '0.00',
+    '-0',
+    '-0.00',
+    '0.01',
+    '0.000000000000000000000000000000000000000001',
+    '-0.01',
+    '-0.000000000000000000000000000000000000000001',
+    '1',
+    '1.0000',
+    '1.0001',
+    '100000.000000000000000000000000000000000000000001',
+    '-1',
+    '-1.0000',
+    '-1.0001',
+    '-100000.000000000000000000000000000000000000000001',
+];
+
+foreach ($nums as $num) {
+    echo str_pad("$num", 50, ' ', STR_PAD_LEFT), ' => ', bcfloor($num), "\n";
+}
+?>
+--EXPECT--
+0 => 0
+                                              0.00 => 0
+                                                -0 => 0
```

Comment:
I wonder whether the result should be `-0` ? (Similarly for cases below and for bcceil)
It would be consistent with: `var_dump(floor(-0.0));`
Genuine question, I don't know the answer for sure.

Reply:
BCMath does not currently support `-0`.

https://github.com/php/php-src/blob/b06c95b6310b07ad4770648d0fb05244930a7092/ext/bcmath/libbcmath/src/num2str.c#L45
https://3v4l.org/HteIO

It seems worth supporting, but I'd like to keep it in a separate scope from this pull request.

Reply:
Okay, I understand.